pkg/endpointmanager/manager.go

```diff
@@ -343,3 +344,18 @@ func AddEndpoint(owner endpoint.Owner, ep *endpoint.Endpoint, reason string) err
 
 	return nil
 }
+
+// WaitForEndpointsAtPolicyRev waits for all endpoints to be at a given policy
+// revision.
+func WaitForEndpointsAtPolicyRev(ctx context.Context, rev uint64) error {
+	eps := GetEndpoints()
+	for i := range eps {
+		<-eps[i].WaitForPolicyRevision(ctx, rev)
```

Comment:
Oh, I see we close it in `Endpoint.setPolicyRevision` but that isn't ideal, since it relies on something calling setPolicyRevision to handle the timeout case.